### PR TITLE
refactor(users): break users/permissions up

### LINKED ISSUE #397: Refactor server/controllers/users.js into multiple files
The `server/controllers/users.js` file is in a terrible state of disrepair.  The code is not at all up to 2.x standards, and the file is far too long.

I recommend: 
1. Break the file into `users.js` and `permissions.js`.
2. Ensure that exports are at the top of the file, with JSDoc syntax
3. Ensure that old routes (such as `authenticatePin`) are removed.

This kind of refactor is important so that we can keep future refactors small and focused.

### PR DESCRIPTION
**Pull request by jniles**
This commit breaks up the users and permissions controllers into smaller files for easier maintainability.  It updates the documentation where appropriate and uses transactions.

Closes #397 .

---

Thank you for contributing!

Before submitting this pull request, please verify that you have:
- [x] Run your code through JSHint.  [Check out our styleguide](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/docs/STYLEGUIDE.md).
- [x] Run integration tests.
- [x] Run end-to-end tests.
- [x] Accurately described the changes your are making in this pull request.

For a more detailed checklist, [see the official review checklist](https://docs.google.com/document/d/1nupLVLRXgSZJQo_acLgrwvPnN8RukfSiwRhSToj81uU/pub) that this PR will be evaluated against.

Ensuring that the above checkboxes are completed will help speed the review process and help build a stronger application.  Thanks!

### CODE REVIEW

File: server/controllers/admin/users/index.js [hunk 1]
```diff
@@ -0,0 +1,269 @@
+/**
+ * @overview Users
```
Comment (sfount) [inline server/controllers/admin/users/index.js:2]: We should decide as a team if we will be going forward with `feature/index.js` as a pattern. If we are it should be formalised and potentially applied to previous modules. 

One disadvantage is fuzzy searches for file names don't work well as everything is called index.js, this isn't a reason to discard the idea though.

**Comment by sfount** [comment]
@jniles There are conflicts in this branch - the code LGTM.

File: server/controllers/admin/users/index.js (hunk 1 above)
Comment (jniles) [inline server/controllers/admin/users/index.js:2]: True - depends on the fuzzy search algorithm I guess.  In theory, `users/index.js` should be similar to `users.js`, just with extra stuff in the middle.

I think the benefits of using `index.js` outweigh the negatives - but would be happy to have a discussion about it.

**Comment by jniles** [comment]
Superseded by #422.

**Pull request closed without merging**